React/Profiler/RCTProfile.m

```diff
@@ -42,8 +42,7 @@
 
 #pragma mark - Variables
 
-// This is actually a BOOL - but has to be compatible with OSAtomic
-static volatile uint32_t RCTProfileProfiling;
+static atomic_bool RCTProfileProfiling = ATOMIC_VAR_INIT(NO);
```

Comment:
Our internal setup is to have `-fno-exceptions` for most objective-c code and opt-in to the codepaths that need it. Maybe we can do that here too?